xbmc/platform/linux/peripherals/PeripheralBusUSBLibUdev.cpp

```diff
@@ -72,40 +72,25 @@ CPeripheralBusUSB::CPeripheralBusUSB(CPeripherals& manager) :
 
   m_udev          = NULL;
   m_udevMon       = NULL;
-
-  if (!(m_udev = udev_new()))
-  {
-    CLog::Log(LOGERROR, "%s - failed to allocate udev context", __FUNCTION__);
-    return;
-  }
-
-  /* set up a devices monitor that listen for any device change */
-  m_udevMon = udev_monitor_new_from_netlink(m_udev, "udev");
-
-  /* filter to only receive usb events */
-  if (udev_monitor_filter_add_match_subsystem_devtype(m_udevMon, "usb", NULL) < 0)
-  {
-    CLog::Log(LOGERROR, "Could not limit filter on USB only");
-  }
-
-  udev_monitor_enable_receiving(m_udevMon);
-
-  CLog::Log(LOGDEBUG, "%s - initialised udev monitor", __FUNCTION__);
 }
 
 CPeripheralBusUSB::~CPeripheralBusUSB(void)
 {
   StopThread(true);
-  udev_monitor_unref(m_udevMon);
-  udev_unref(m_udev);
 }
 
 bool CPeripheralBusUSB::PerformDeviceScan(PeripheralScanResults &results)
 {
   struct udev_enumerate *enumerate;
   struct udev_list_entry *devices, *dev_list_entry;
   struct udev_device *dev(NULL), *parent(NULL);
-  enumerate = udev_enumerate_new(m_udev);
```

Comment:
@FernetMenta this was the only connection to the internal base class if called from outside. So I just removed it.